di-ipv-core-stub/src/main/java/uk/gov/di/ipv/stub/core/handlers/BasicAuthHandler.java

```diff
@@ -0,0 +1,54 @@
+package uk.gov.di.ipv.stub.core.handlers;
+
+import spark.Filter;
+import spark.Request;
+import spark.Response;
+import spark.Spark;
+import uk.gov.di.ipv.stub.core.config.CoreStubConfig;
+
+import java.util.Base64;
+
+public class BasicAuthHandler {
+    private static int NUMBER_OF_AUTHENTICATION_FIELDS = 2;
+    private static final String AUTHORIZATION_HEADER = "Authorization";
+    private static final String AUTHORIZATION_TYPE = "Basic";
+
+    public BasicAuthHandler() {
+        CoreStubConfig.getUserAuth();
+    }
+
+    public Filter authFilter =
+            (Request request, Response response) -> {
+                if (!request.headers().contains(AUTHORIZATION_HEADER) || !authenticated(request)) {
+                    response.header("WWW-Authenticate", AUTHORIZATION_TYPE);
+                    Spark.halt(401, "Not Authenticated");
+                }
+            };
```

Comment:
I don't know business requirement. I think, we can add some logs after successfully logging.